web_notify/static/src/js/web_client.js

```diff
@@ -100,5 +97,5 @@ WebClient.include({
         }
     }
 });
-
+return WebClient;
```

Comment:
This is not needed.

Reply:
Here are some things i found problematic - maybe i am wrong - but after my changes it did work as expected... ;-)

* start_polling function in web_client.js this function is ok if you use only the web_notify module. The web_auto_refresh module did used the same function (also with include on WebClient) - so you get unexpected results (last come first serve). So in my opinion it is better to not use too generic function names - or extend directly they function from where it got called.
* function bus_notification: same problem here - the name is too generic - other modules also use the name (as the docs also use this name - but for own module - not when directly included in WebClient !) - so the new name is not the best one - so maybe we could prefix such functions - lets call it _web_notify_bus_notification
* About the return WebClient - you are correct - this is not needed with include

Reply:
OK, I understand the problem.

Remember to follow the same logic in the other PR then, and please remove the return.

Reply:
@Yajo Out of curiosity but why do you say that the `return` should be removed? If we want to extend the specific JS provided by this addon on the WebClient, we should return WebClient. In this way we'll be able to write
```javascript
odoo.define('my_extended_web_notify.WebClient', function (require) {
    var WebClient = require('web_notify.WebClient');
    ...
});
```

Do I miss something?

Reply:
It's just not necessary. When calling `.include()`, you alter the original object's protoype, so further calls to the original object are the same as those for the returned object. If we were creating a subclass (through `.extend()`), then yes, it should be returned to be able to extend it further.

Your code and this code are exactly the same:

```js
odoo.define('my_extended_web_notify.WebClient', function (require) {
    require('web_notify.WebClient');
    var WebClient = require('web.WebClient'),
    ...
});
```

It's not actually *an error* to return that class, but it's simply not needed as you can see. And if you don't need something, it's better not to have it IMHO.

Reply:
OK thank you for the explanation. I thought that `require' was also used to ensure the loading order of the JS but on the other hand it's the extend of asset.backend  that should do it.

Reply:
require is async, so the order does not matter AFAIK.

Reply:
no convinced but not an expert... :smirk: